taker-frontend/src/components/History.tsx

```diff
@@ -93,13 +93,18 @@ const CfdDetails = ({ cfd, connectedToMaker, displayCloseButton }: CfdDetailsPro
             isForceCloseButton={true}
         />;
 
+    let failedCfd = [StateKey.REJECTED, StateKey.SETUP_FAILED].includes(cfd.state.key);
+
     let profitColors = !cfd || !cfd.profit_btc || cfd.profit_btc === 0
-            || [StateKey.REJECTED, StateKey.SETUP_FAILED].includes(cfd.state.key)
+            || failedCfd
         ? { light: "gray.500", dark: "gray.400" }
         : cfd.profit_btc > 0
         ? { light: "green.600", dark: "green.300" }
         : { light: "red.600", dark: "red.300" };
 
+    let profitLabel = isClosed(cfd) ?
+        failedCfd ? "Missed P/L 😭" : "Realized P/L" : "Unrealized P/L";
```

Comment:
😂😂😂

Definitely not enough emojis in this codebase.

Reply:
/me looking for a usecase for 💩

Reply:
Note: Making these things components with early return might be easier to understand.